scripts/restore.sh

```diff
@@ -0,0 +1,120 @@
+RESTORE_PATH="/palworld/Pal"
+
+# Copy the save file before restore temporary path
+TMP_SAVE_PATH="/palworld/backups/restore-"$(date +"%Y-%m-%d_%H-%M-%S")
+
+# shellcheck disable=SC2317
+term_error_handler() {
+    echo "An error occurred during server shutdown."
+    exit 1
+}
+
+# shellcheck disable=SC2317
+restore_error_handler() {
+    printf "\033[0;31mAn error occurred during restore.\033[0m\n"
+    if [ -d "$TMP_SAVE_PATH/Saved" ]; then
+        read -rp "I have a backup before recovery can proceed. Do you want to recovery it? (y/n): " RUN_ANSWER
+        if [[ $RUN_ANSWER == "y" ]] || [[ $RUN_ANSWER == "Y" ]]; then
+            rm -rf "$RESTORE_PATH/Saved"
+            mv "$TMP_SAVE_PATH/Saved" "$RESTORE_PATH"
+            printf "\e[0;32mRecovery complete.\e[0m\n"
+        fi
+    fi
+
+    echo "Clean up the temporary directory."
+    rm -rf "$TMP_PATH"
+    rm -rf "$TMP_SAVE_PATH"
+
+    exit 1
+}
+
+if [ "${RCON_ENABLED}" != true ]; then
+    echo "RCON is not enabled. Please enable RCON to use this feature."
+    exit 1
+fi
+
+# Show up backup list
+echo "Backup List:"
+select BACKUP_FILE in "$BACKUP_DIRECTORY_PATH"/*; do
+    if [ -n "$BACKUP_FILE" ]; then
+        echo "Selected backup: $BACKUP_FILE"
+        break
+    else
+        echo "Invalid selection. Please try again."
+    fi
+done
+
+if [ -f "$BACKUP_FILE" ]; then
+    echo "Do you want to continue with the command?"
+    read -rp "When you run it, the server will be stopped and the recovery will proceed. (y/n): " RUN_ANSWER
+    if [[ $RUN_ANSWER == "y" ]] || [[ $RUN_ANSWER == "Y" ]]; then
+        printf "\e[0;32m*****STARTING PROCESS*****\e[0m\n"
+        # Shutdown server
+        trap 'term_error_handler' ERR
+
+        if [ "${RCON_ENABLED}" = true ]; then
+            printf "\e[0;32m*****SHUTDOWN SERVER*****\e[0m\n"
+            rcon-cli -c /home/steam/server/rcon.yaml save
+            rcon-cli -c /home/steam/server/rcon.yaml "shutdown 1"
+        else
+            echo "RCON is not enabled. Please enable RCON to use this feature. Unable to restore backup."
+            exit 1
+        fi
+        printf "\e[0;32mShutdown complete.\e[0m\n"
+
+        trap - ERR
+
+        trap 'restore_error_handler' ERR
+        
+        printf "\e[0;32m*****START RESTORE*****\e[0m\n"
+
+        # Recheck the backup file
+        if [ -f "$BACKUP_FILE" ]; then
+            # Copy the save file before restore
+            if [ -d "$RESTORE_PATH/Saved" ]; then
+                echo "Saves the current state before the restore proceeds."
+                echo "$TMP_SAVE_PATH"
+                mkdir -p "$TMP_SAVE_PATH"
+                \cp -rf "$RESTORE_PATH/Saved" "$TMP_SAVE_PATH/Saved"
+
+                while [ ! -d "$TMP_SAVE_PATH/Saved" ]; do
+                    sleep 1
+                done
+
+                printf "\e[0;32mSave complete.\e[0m\n"
+            fi
+            
+            # Create tmp directory
+            TMP_PATH=$(mktemp -d -p "/palworld/backups")
```

Comment:
I would add a chmod in so the file is owned by steam not root. 
In the event of a container kill or sigterm on restore it will remain.
While it shouldn't be needed since the container `chown -R steam:steam /palworld /home/steam/` it prevents having a directory owned by root from existing. 

```suggestion
            TMP_PATH=$(mktemp -d -p "/palworld/backups")
            if [ "$(id -u)" -eq 0 ]; then
                chown steam:steam "$TMP_PATH"
            fi
```

Reply:
This folder is likewise a temporary directory for extracting the backup `.tar` file. 

It will be deleted upon completion of the restore process or if an error occurs.

Will we ever need to specify ownership?